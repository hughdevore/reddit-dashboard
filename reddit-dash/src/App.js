import React, { Component, Fragment } from 'react';
import { Layout, Skeleton } from 'antd';
import Article from './components/Article/Article';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      article: {},
      isLoading: false,
    };
  }

  async fetchArticle() {
    let response = await fetch('https://gist.githubusercontent.com/mkg0/6a4dca9067ad7a296204e7c9ecd977b0/raw/0b1ec16580ea1e970a73f5c85563c22631be7ad7/unpopularopinion-dataset.json');
    return response.json();
  }

  componentDidMount() {
    this.setState({ isLoading: true });
  
    // Fetch the data for the article and comments.
    this.fetchArticle()
    .then(article => {
      setTimeout( () => {
        console.log(article);
        this.setState({ article, isLoading: false })
      }, 750 );

    })
    .catch(err => {
      console.log(err);
    });

  }

  render() {
    const { article, isLoading } = this.state;

    return (
      <Fragment>
        <Layout
          style={{
            backgroundColor: 'rgb(222, 228, 234)',
            padding: '7em 10em',
            height: '75vh',
          }}
        >
          <Layout
            style={{
              backgroundColor: 'rgb(249, 249, 249)',
              padding: '3em 6em',
              borderRadius: '8px',
              boxShadow: '0 0 10px gray',
              fontFamily: 'verdana'
            }}
          >
            <Skeleton active={true} loading={isLoading}>
              <Article article={article} />
            </Skeleton>
          </Layout>
        </Layout>
      </Fragment>
    );
  }
  
}

export default App;
