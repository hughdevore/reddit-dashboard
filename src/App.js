import React, { Component, Fragment } from 'react';
import { Layout, Skeleton } from 'antd';
import 'antd/lib/skeleton/style/css';
import Article from './components/Article/Article';
import fetchArticle from './utils/fetchArticle';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      article: {},
      isLoading: false,
    };
  }

  componentDidMount() {
    this.setState({ isLoading: true });
    // Fetch the data for the article and comments.
    fetchArticle()
      .then(article => {
        setTimeout(() => {
          this.setState({ article, isLoading: false })
        }, 750);
      })
      .catch(err => {
        console.log(err);
      });
  }

  render() {
    const { article, isLoading } = this.state;
    return (
      <Fragment>
        <Layout
          style={{
            backgroundColor: 'rgb(222, 228, 234)',
            padding: '7em 10em',
            minHeight: '100vh'
          }}
        >
          <Layout
            style={{
              backgroundColor: 'rgb(249, 249, 249)',
              padding: '3em 6em',
              borderRadius: '15px',
              boxShadow: '0 0 10px gray',
              fontFamily: 'verdana'
            }}
          >
            <Skeleton active={true} loading={isLoading}>
              <Article article={article} />
            </Skeleton>
          </Layout>
        </Layout>
      </Fragment>
    );
  }
}

export default App;
