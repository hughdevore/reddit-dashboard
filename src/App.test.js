import React from "react";
import { unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';
import App from "./app";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('The App component ', () => {
  it('renders correctly.', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});