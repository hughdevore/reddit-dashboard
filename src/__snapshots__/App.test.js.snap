// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App component  renders correctly. 1`] = `
<section
  className="ant-layout"
  style={
    Object {
      "backgroundColor": "rgb(222, 228, 234)",
      "minHeight": "100vh",
      "padding": "7em 10em",
    }
  }
>
  <section
    className="ant-layout"
    style={
      Object {
        "backgroundColor": "rgb(249, 249, 249)",
        "borderRadius": "15px",
        "boxShadow": "0 0 10px gray",
        "fontFamily": "verdana",
        "padding": "3em 6em",
      }
    }
  >
    <div
      className="ant-skeleton ant-skeleton-active"
    >
      <div
        className="ant-skeleton-content"
      >
        <h3
          className="ant-skeleton-title"
          style={
            Object {
              "width": "38%",
            }
          }
        />
        <ul
          className="ant-skeleton-paragraph"
        >
          <li
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <li
            style={
              Object {
                "width": undefined,
              }
            }
          />
          <li
            style={
              Object {
                "width": "61%",
              }
            }
          />
        </ul>
      </div>
    </div>
  </section>
</section>
`;
